File: 2800-minimum-string-length-after-removing-substrings/minimum-string-length-after-removing-substrings.py
class Solution:
    def minLength(self, s: str) -> int:
        while True:
            if "AB" in s:
                s = s.replace("AB", "")
            elif "CD" in s:
                s = s.replace("CD", "")
            else :
                break
        return len(s)

File: 2800-minimum-string-length-after-removing-substrings/minimum-string-length-after-removing-substrings.cpp
class Solution {
public:
    int minLength(string s) {
        stack<char> stack;

        for (auto chr : s) {
            if (!stack.empty() && (chr == 'B' && stack.top() == 'A') ||
                !stack.empty() && chr == 'D' && stack.top() == 'C') {
                stack.pop();
            } else {
                stack.push(chr);
            }
        }
        return stack.size();
    }
};